boot/bootutil/src/image_validate.c

```diff
@@ -631,7 +634,12 @@ bootutil_img_validate(struct boot_loader_state *state,
             if (rc) {
                 goto out;
             }
+#if !defined(MCUBOOT_BYPASS_KEY_MATCH)
             key_id = bootutil_find_key(buf, len);
+#else
+            /* There is only one key */
+            key_id = 0;
```

Comment:
consider function returning 0 for MCUBOOT_BYPASS_KEY_MATCH:
``` C
static int bootutil_find_key(buf, len)
{
  (void) buf;
   (void) len;
   return 0;
}
```

rationale: less #ifdefs cutting function.